pkg/device_plugin/device_plugin.go

```diff
@@ -308,32 +309,46 @@ func getDeviceName(deviceID string) string {
 	}
 	defer file.Close()
 
+	// Find beginning of NVIDIA device list
 	scanner := bufio.NewScanner(file)
 	for scanner.Scan() {
 		line := scanner.Text()
-		line = strings.TrimSpace(line)
-		if line != "" && strings.Contains(line, deviceID) {
-			splits := strings.Split(line, deviceID)
-			if len(splits) != 2 {
-				log.Printf("Error processing pci.ids file at line: %s", line)
-				return deviceName
-			}
-			// fix matching
-			if splits[0] != "" {
-				continue
-			}
-			deviceName = strings.TrimSpace(splits[1])
-			deviceName = strings.ToUpper(deviceName)
-			deviceName = strings.Replace(deviceName, "/", "_", -1)
-			deviceName = strings.Replace(deviceName, ".", "_", -1)
-			reg, _ := regexp.Compile("\\s+")
-			deviceName = reg.ReplaceAllString(deviceName, "_") // Replace all spaces with underscore
-			reg, _ = regexp.Compile("[^a-zA-Z0-9_.]+")
-			deviceName = reg.ReplaceAllString(deviceName, "") // Removes any char other than alphanumeric and underscore
+		if strings.HasPrefix(line, "10de") {
```

Comment:
Why are there two `scanner.Scan()` calls?  Seems like the `for scanner.Scan() { .. ` code below does what you need.

Reply:
Without the first for loop, we are only considering the device id, not vendor id.

Consider the below example (which exists now in the updated pci.ids file):
```
8086  Intel Corporation
	2331  DH89xxCC Chap Counter
10de NVIDIA Corporation
	2331 GH100 [H100 PCIe]
```
The Intel device shares the same device id as H100 PCIe. The first for loop ensures we only start looking for a matching device id once we have reached the section containing NVIDIA devices.

Reply:
Ok I would put this behind a function then so it reads a bit better.

```go
const nvidiaVendorID = 10de
locateVendor(nvidiaVendorID)

for scanner.Scan() { ... }
```

Reply:
Updated as you have suggested.